### PR TITLE
Harmony: alerts

### LINKED ISSUE #766: Harmony: alerts
Should probably be a separate component
Design: https://www.figma.com/file/3xsUrRBsEJTBrJ27gMhSGZ/Taskany-(Goals)-New?type=design&node-id=2923-55748&mode=design&t=bSkmaS6LpUO5NGd8-4

Should probably be a base component for a tooltip, just without ref support

### PR DESCRIPTION
**Pull request by Katrin-kudryash**
## PR includes

- [x] Feature

## Related issues

Resolve #766

### CODE REVIEW

**Comment by Katrin-kudryash** [comment]
Untitled 2

**Comment by Katrin-kudryash** [comment]
Untitled

File: src/harmony/Alert/Alert.module.css [hunk 1]
```diff
@@ -0,0 +1,45 @@
+.Alert {
+    display: flex;
+    align-items: center;
+    justify-content: start;
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.module.css:4]: Does it not default?

File: src/harmony/Alert/Alert.module.css [hunk 2]
```diff
@@ -0,0 +1,45 @@
+.Alert {
+    display: flex;
+    align-items: center;
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.module.css:3]: Why padding is not enough?

File: src/harmony/Alert/Alert.module.css [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    justify-content: start;
+    border-radius: var(--radius-m);
+    padding: var(--gap-sm) var(--gap-sm);
+    gap: var(--gap-s);
+    width: 22.25rem;
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.module.css:8]: What the size and why?

File: src/harmony/Alert/Alert.module.css [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+.Alert_default {
+    color: var(--gray-900);
+    background: var(--gray-900);
+    z-index: 1;
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.module.css:14]: Looks like copy/paste from Figma?

File: src/harmony/Alert/Alert.module.css [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    color: var(--text-primary);
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.module.css:15]: Double color? What is right?

File: src/harmony/Alert/Alert.module.css [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+.Alert_danger {
+    color: var(--danger-300);
+    background: var(--danger-900);
+    z-index: 1;
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.module.css:21]: ?

File: src/harmony/Alert/Alert.module.css [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+.Alert_warning {
+    color: var(--warn-300);
+    background: var(--warn-900);
+    z-index: 1;
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.module.css:27]: ?

File: src/harmony/Alert/Alert.module.css [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+.Text {
+    font-size: var(--font-size-s);
+    line-height: var(--line-height-s);
+}
+
+.Text_default {
+    color: var(--gray-300);
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.module.css:36]: You are using same style for Alert container

File: src/harmony/Alert/Alert.module.css [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+.Text_danger {
+    color: var(--danger-300);
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.module.css:40]: ?

File: src/harmony/Alert/Alert.module.css [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+.Text_warning {
+    color: var(--warn-300);
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.module.css:44]: ?

File: src/harmony/Alert/Alert.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+            flexDirection: 'column',
+            gap,
+            justifyContent: 'start',
+            width: '100%',
+        }}
+    >
+        {children}
+    </div>
+);
+const Title = ({ children }: { children: ReactNode }) => (
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.stories.tsx:20]: Use Text component instead

File: src/harmony/Alert/Alert.stories.tsx [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+    <div style={{ fontSize: 36, marginBottom: 20, fontWeight: 'bold' }}>{children}</div>
+);
+
+const Text = ({ children }: { children: ReactNode }) => <span style={{ fontSize: 16 }}>{children}</span>;
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.stories.tsx:24]: Use Text component instead

File: src/harmony/Alert/Alert.stories.tsx [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+    return (
+        <>
+            <Title>
+                Allerts <Text>Default / Danger / Warning </Text>
+            </Title>
+            <Layout>
+                <Alert
+                    view="default"
+                    icon={<IconInfoCircleOutline size={15} />}
+                    text={'The selected date has already passed'}
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.stories.tsx:43]: Do u need help with prettier locally?

File: src/harmony/Alert/Alert.tsx [hunk 14]
```diff
@@ -0,0 +1,35 @@
+import React, { ReactNode } from 'react';
+import cn from 'classnames';
+
+import { Text } from '../../harmony/Text/Text';
+
+import s from './Alert.module.css';
+// import { IconExclamationCircleOutline, IconInfoCircleOutline } from '@taskany/icons';
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.tsx:7]: ?

File: src/harmony/Alert/Alert.module.css [hunk 15]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+.Alert_warning {
+    color: var(--warn-300);
+    background: var(--warn-900);
+    z-index: 1;
+}
+
+.Text {
+    font-size: var(--font-size-s);
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.module.css:31]: Use Text component props

File: src/harmony/Alert/Alert.tsx [hunk 16]
```diff
@@ -0,0 +1,35 @@
...
+    text: string;
+    icon: ReactNode;
+}
+const viewMap = {
+    default: s.Alert_default,
+    danger: s.Alert_danger,
+    warning: s.Alert_warning,
+};
+
+const TextColorMap = {
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.tsx:20]: Do u really need this one? Why .Alert style is not enough?

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/harmony/Alert/Alert.module.css (hunk 1 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.module.css]: yes, fixed it

File: src/harmony/Alert/Alert.module.css (hunk 2 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.module.css:3]: Used to align icons

File: src/harmony/Alert/Alert.module.css (hunk 3 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.module.css]: deleted it

File: src/harmony/Alert/Alert.module.css (hunk 4 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.module.css]: fixed

File: src/harmony/Alert/Alert.module.css (hunk 5 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.module.css:11]: deleted it

File: src/harmony/Alert/Alert.module.css (hunk 7 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.module.css]: deleted it

File: src/harmony/Alert/Alert.module.css (hunk 8 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.module.css]: fixed

File: src/harmony/Alert/Alert.module.css (hunk 9 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.module.css]: fixed

File: src/harmony/Alert/Alert.module.css (hunk 10 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.module.css]: fixed

File: src/harmony/Alert/Alert.stories.tsx (hunk 12 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.stories.tsx:22]: Text doesn't have a prop style

File: src/harmony/Alert/Alert.stories.tsx (hunk 13 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.stories.tsx]: fixed

File: src/harmony/Alert/Alert.tsx (hunk 14 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.tsx]: fixed it

File: src/harmony/Alert/Alert.module.css (hunk 15 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.module.css]: done

File: src/harmony/Alert/Alert.tsx (hunk 16 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.tsx]: fixed

File: src/harmony/Alert/Alert.tsx [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+import { IconExclamationCircleOutline, IconInfoCircleOutline } from '@taskany/icons';
+
+import { Text } from '../../harmony/Text/Text';
+
+import s from './Alert.module.css';
+
+interface AlertProps extends React.HTMLAttributes<HTMLDivElement> {
+    view: 'default' | 'danger' | 'warning';
+    text: string;
+    icon: ReactNode;
```
Comment (tsumo) [inline src/harmony/Alert/Alert.tsx:12]: icon should be optional

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/harmony/Alert/Alert.tsx [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+const iconMap = {
+    default: <IconInfoCircleOutline size={15} />,
+    danger: <IconExclamationCircleOutline size={15} />,
+    warning: <IconExclamationCircleOutline size={15} />,
+};
+
+export const Alert: React.FC<AlertProps> = ({ text, icon, className, view, ...props }) => {
+    return (
+        <div className={cn(s.Alert, viewMap[view], className)} {...props}>
+            {iconMap[view]}
```
[the remark this answers is not in the document]
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.tsx]: done

File: src/harmony/Alert/Alert.tsx (hunk 17 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.tsx]: done

File: src/harmony/Alert/Alert.stories.tsx (hunk 11 above)
Comment (awinogradov) [reply src/harmony/Alert/Alert.stories.tsx:18]: Exactly, but it has size and weight props. Try to check storybook first.

File: src/harmony/Alert/Alert.stories.tsx (hunk 12 above)
Comment (awinogradov) [reply src/harmony/Alert/Alert.stories.tsx:22]: I'm not sure

File: src/harmony/Alert/Alert.tsx [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+const iconMap = {
+    default: <IconInfoCircleOutline size={15} />,
+    danger: <IconExclamationCircleOutline size={15} />,
+    warning: <IconExclamationCircleOutline size={15} />,
+};
+
+export const Alert: React.FC<AlertProps> = ({ text, icon, className, view, ...props }) => {
+    return (
+        <div className={cn(s.Alert, viewMap[view], className)} {...props}>
+            {icon || iconMap[view]}
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.tsx:29]: `nullable(icon, (i) => i, iconMap[view])`

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/harmony/Alert/Alert.stories.tsx [hunk 20]
```diff
@@ -0,0 +1,46 @@
...
+    component: Alert,
+};
+
+export default meta;
+
+const Alerts = () => {
+    return (
+        <>
+            <Title>
+                Allerts <Text>Default / Danger / Warning </Text>
```
Comment (tsumo) [inline src/harmony/Alert/Alert.stories.tsx:35]: Typo "allerts"

File: src/harmony/Alert/Alert.module.css [hunk 21]
```diff
@@ -0,0 +1,22 @@
+.Alert {
+    display: flex;
+    align-items: center;
+    border-radius: var(--radius-m);
+    padding: var(--gap-sm) var(--gap-sm);
```
Comment (tsumo) [inline src/harmony/Alert/Alert.module.css:5]: One var is enough, since they are the same

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/harmony/Alert/Alert.module.css (hunk 21 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.module.css]: done

File: src/harmony/Alert/Alert.stories.tsx (hunk 20 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.stories.tsx]: done

File: src/harmony/Alert/Alert.tsx (hunk 19 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.tsx]: ok, thanks

File: src/harmony/Alert/Alert.stories.tsx [hunk 22]
```diff
@@ -0,0 +1,58 @@
...
+    title: '@Harmony/Alert',
+    component: Alert,
+};
+
+export default meta;
+
+const Alerts = () => {
+    return (
+        <>
+            <Text size="xxl" as="label" weight="bold">
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.stories.tsx:19]: https://developer.mozilla.org/ru/docs/Web/HTML/Element/label

File: src/harmony/Alert/Alert.stories.tsx [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+                <Alert
+                    view="default"
+                    icon={<IconInfoCircleOutline size={15} />}
+                    text="The selected date has already passed"
+                />
+                <Alert
+                    view="danger"
+                    icon={<IconExclamationCircleOutline size={15} />}
+                    text="An error has occurred, please try again later"
+                    color=""
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.stories.tsx:45]: ?

File: src/harmony/Alert/Alert.stories.tsx [hunk 24]
```diff
@@ -0,0 +1,58 @@
...
+                />
+                <Alert
+                    color=""
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.stories.tsx:48]: ?

**Review by awinogradov — CHANGES_REQUESTED** [review]

File: src/harmony/Alert/Alert.stories.tsx (hunk 23 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.stories.tsx]: fixed

File: src/harmony/Alert/Alert.stories.tsx (hunk 24 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.stories.tsx]: fixed

File: src/harmony/Alert/Alert.stories.tsx (hunk 22 above)
Comment (Katrin-kudryash) [reply src/harmony/Alert/Alert.stories.tsx]: fixed

**Comment by Katrin-kudryash** [comment]
Untitled 3

**Comment by Katrin-kudryash** [comment]
https://github.com/taskany-inc/bricks/assets/81121562/b24bf409-9125-4912-afda-2c37dac5d5fe

File: src/harmony/Alert/Alert.tsx [hunk 25]
```diff
@@ -0,0 +1,34 @@
...
+    icon?: ReactNode;
+}
+const viewMap = {
+    default: s.Alert_default,
+    danger: s.Alert_danger,
+    warning: s.Alert_warning,
+};
+
+const iconMap = {
+    default: <IconInfoCircleOutline size={'s'} />,
```
Comment (awinogradov) [inline src/harmony/Alert/Alert.tsx:22]: 🤦🏼‍♂️

**Review by awinogradov — CHANGES_REQUESTED** [review]

**Review by awinogradov — APPROVED** [review]

**Review by tsumo — APPROVED** [review]

**Pull request merged**